src/settings/FileExtensions/FileExtensions.js

```diff
@@ -333,48 +246,42 @@ export class FileExtensions extends Component {
     return (
       <IntlConsumer>
         {intl => (
-          <Fragment>
-            <div
-              className={sharedCss.container}
-              data-test-file-extensions
-            >
-              <SearchAndSort
-                objectName="file-extensions"
-                finishedResourceName="fileExtensions"
-                parentResources={resources}
-                parentMutator={mutator}
-                initialResultCount={INITIAL_RESULT_COUNT}
-                resultCountIncrement={RESULT_COUNT_INCREMENT}
-                searchLabelKey="ui-data-import.settings.fileExtensions.title"
-                resultCountMessageKey="ui-data-import.settings.fileExtensions.count"
-                resultsLabel={label}
-                defaultSort="extension"
-                actionMenu={this.renderActionMenu}
-                resultsFormatter={resultsFormatter(intl, searchTerm)}
-                visibleColumns={this.visibleColumns}
-                columnMapping={{
-                  extension: intl.formatMessage({ id: 'ui-data-import.settings.fileExtension.extension' }),
-                  importBlocked: intl.formatMessage({ id: 'ui-data-import.settings.fileExtension.blockImport' }),
-                  dataTypes: intl.formatMessage({ id: 'ui-data-import.settings.fileExtension.dataTypes' }),
-                  updated: intl.formatMessage({ id: 'ui-data-import.updated' }),
-                  updatedBy: intl.formatMessage({ id: 'ui-data-import.updatedBy' }),
-                }}
-                columnWidths={this.columnWidths}
-                fullWidthContainer={this.fullWidthContainer}
-                ViewRecordComponent={ViewFileExtension}
-                EditRecordComponent={FileExtensionForm}
-                newRecordInitialValues={newRecordInitialValues}
-                showSingleResult={showSingleResult}
-                onCreate={this.createRecord}
-                onEdit={this.editRecord}
-                onDelete={this.deleteRecord}
-                handleCreateSuccess={this.handleUpdateRecordSuccess}
-                handleEditSuccess={this.handleUpdateRecordSuccess}
-              />
-            </div>
-            <div
-              className={sharedCss.fullWidthAndHeightContainer}
-              ref={this.createFullWidthContainerRef}
+          <div
+            className={sharedCss.container}
+            data-test-file-extensions
+          >
+            <SettingPage
```

Comment:
The approach is worthy. I think it makes sense to improve it by render prop pattern so the prop drilling would be avoided.
```
  <SettingsPage>
    {(props) => { // props needed for searchAndSort
       return (
         <Fragment>
           <SearchAndSort
             {...searchAndSortProps} // searchAndSortProps now should not be part of `SettingsPage` component and thus not be presented in its prop types
           >
           </SearchAndSort>
           ...
         <Fragment>
       )
    }}
  </SettingsPage>
```